Nuclear operatives can now be other species. Pref toggle to always be human as operative

Rhials opened the pull request:
## About The Pull Request

![image](https://github.com/tgstation/tgstation/assets/28870487/cbfbf18a-b4cb-44d1-97ef-430fc4404128)

When #16075 was created in the ancient year of 2016, it was a stopgap solution to plasmamen spawning in with a regular nukie outfits and burning to death. As a result, nukies have been human-only for the past seven years. I come to you today to remove the stopgap, and once again enable non-human nukies.

This was approved in the past (#59590), and I'm assuming that approval is still valid. Let's take a look at how things work:

Spawning in as a Nuclear Operative (leader included), Lone Operative, Clown Operative, or Reinforcement will now set your species to the one if your prefs. That means lizard, cat, plasma, and ethereal nukies are all on the table now.

Plasmamen now have their own syndicate envirohelm/suit combo. It doesn't look fantastic (it's just a palette swapped roboticists envirosuit) but I think it's serviceable at least. When the operative antag datum is given, and the owner is a plasmamen, they will instead be given an alternate outfit with these equipped. Loneop plasmamen will instead be given a MODsuit with the plasma stabilizer MODule installed. These MODules also spawn in the firebase lockers in case they're needed.

**But what about the AI?** Fortunately for the balancebrains in the crowd, there is a preference (enabled by default!) to always roll human as an operative. The AI still has to reasonably assume that nukies are human until proven otherwise so please don't think this will affect the rules of engagement for anyone not _directly opting into it._

![image](https://github.com/tgstation/tgstation/assets/28870487/f4ba8e6d-bfb8-40cb-abcb-95e8ea224a95)

(This pref will only appear if you have a nukie job enabled in the antag menu!)

Huge thanks to Melbert who let me peek at his code on a branch trying to do the same thing. My approach ended up being entirely on the antag datum, with nothing on the job datum.
## Why It's Good For The Game

Rectifies a curious change made long long ago. Expands customization to nukies while not affecting balance for those who care.
## Changelog
:cl: Rhials
add: Nuclear operatives have expanded the scope of their hiring practices. Members of all species can now roll nukie!
qol: The preferences menu now has an option to always be human when being selected for a nuclear operative role. Check it out!
image: Plasmaman operative envirosuits.
/:cl:

Rhials committed:
first steps

creates the premade modsuit, equipment and outfits

im going to try hooking the outfit distribution through the antag datums instead of the job datums

Rhials committed:
job outfits, comments, makes things work

Rhials committed:
adds the proper suits

presently loneops are always human bc of how the identity is assigned maybe ill change that

still need worn sprite for nukie envirosuit

Rhials committed:
fixes the suit

Rhials committed:
some more fixes

wanna hear something funny?

every time I spawned in a plasmanukie, I had to turn up the plasma pressure or else I would suffocate

I was confused for a while and y'know what I found caused it?

the syndie gas masks I was "helpfully" spawning plasmemes with were filtering(?) the gas out and causing the wearer to not breathe in anything useful

wow!

Rhials committed:
reinforcements and loneop fixe

Rhials committed:
makes the reinforcement beacon not kill you I hope

the more i clean up this code, the uglier it looks

Rhials committed:
hmmmmmm

everything goes through the antag datum, no need to check species anywhere as long as both nukeop_outfit and plasmaman_outfit have been changed to their equivalents

Jackal-boop commented:
this is good

mc-oofert commented:
if my leader was a nonhuman i would probably call it a guaranteed flukeop

Livrah commented:
Like flukie teammates wasn't bad enough already, some races take extra dmg.

Ticklezcat commented:
> Like flukie teammates wasn't bad enough already, some races take extra dmg.

Securities weakest lasers vs. The Syndicate's strongest moff.

DaydreamIQ commented:
As a human-only player, this is pretty funny. Plasmeme ops might be weaker but it makes their win all the stronger.

MrMelbert commented on code/modules/antagonists/nukeop/nukeop.dm:
```diff
@@ -35,9 +41,15 @@
+	if(!operative.client || operative.client.prefs.read_preference(/datum/preference/toggle/nuke_ops_species))
+		operative.set_species(/datum/species/human)
+
+	if(is_species(operative, /datum/species/plasmaman))
+		operative.equipOutfit(plasmaman_outfit)
```
This should not be a hardcoded thing, it should defer to the species, IE via `pre_equip_species_outfit`. 
If we (or downstreams) wish to add a new species which requires its own unique outfits, we shouldn't have to come back to this to add `isplasmaman, elsif snowflakeman, elseif fishman, else...`

(this is the reason why I used jobs datums, it worked with that proc nicely)

MrMelbert commented on code/modules/antagonists/_common/antag_spawner.dm:
```diff
@@ -157,7 +159,13 @@
 	antag_datum = new()
 	antag_datum.send_to_spawnpoint = FALSE
+
+	if(!our_client.prefs.read_preference(/datum/preference/toggle/nuke_ops_species))
+		var/species_type = our_client.prefs.read_preference(/datum/preference/choiced/species)
+		nukie.set_species(species_type)
```
This feels backwards or unnecessary. 
`safe_transfer_prefs_to` should already set them to their pref species.

bruhlookatthisdood commented:
i feel like you should be able to choose what species you play as

moocowswag commented:
non human nukies do have advantages though, lizard gets tackle resistance so its slightly safer to go desword / shielded modsuit.

plasmameme can try to do a singuloose/tesloose sm delam and try to avoid direct combat all together and just make the station a bad time for the crew.

Plasmameme`s downside is also irrelevant if played safely with shielded modsuit...atleast until bombs come into play.

Ethereal....cant revive crystal because nukies gib themselves on death :(

Don`t moffs see better in the dark?

Rex9001 commented:
> non human nukies do have advantages though, lizard gets tackle resistance so its slightly safer to go desword / shielded modsuit.
> 
> 
> 
> plasmameme can try to do a singuloose/tesloose sm delam and try to avoid direct combat all together and just make the station a bad time for the crew.
> 
> 
> 
> Plasmameme`s downside is also irrelevant if played safely with shielded modsuit...atleast until bombs come into play.
> 
> 
> 
> Ethereal....cant revive crystal because nukies gib themselves on death :(
> 
> 
> 
> Don`t moffs see better in the dark?
> 
> 
> 
> And felinids can catch the lasers for the other nukies like a cat trying to chase a laser pointer that kills them.

You can disable the microbomb implant with the atropine injector in your box

Googles-Hands commented:
Mirror NT's Head of Staff and restrict nuke ops leaders to human only. :)

Rhials committed:
whaddya know it wasnt necessary at all

san7890 commented:
ready for review when dehardcoding is done

necromanceranne commented:
Woo, plasmaman ops!

Besure to have plasmaman ops spawn with plasma stabilizers to put into their modsuits.

SimplySmiley123 commented:
> non human nukies do have advantages though, lizard gets tackle resistance so its slightly safer to go desword / shielded modsuit.
> 
> plasmameme can try to do a singuloose/tesloose sm delam and try to avoid direct combat all together and just make the station a bad time for the crew.
> 
> Plasmameme`s downside is also irrelevant if played safely with shielded modsuit...atleast until bombs come into play.
> 
> Ethereal....cant revive crystal because nukies gib themselves on death :(
> 
> Don`t moffs see better in the dark?
> 
> And felinids can catch the lasers for the other nukies like a cat trying to chase a laser pointer that kills them.

Honestly adds some variety with some downsides to boot.
Like ethereals and plasmamen taking more direct damage and all of the non-humans not being protected from Shitsimov.

Fritzlschnitzel commented:
The syndicate has fallen.

ExcessiveUseOfCobblestone commented:
> Some players play hardcore random as antagonists already, doubt taking a lizard specie is gonna do any worse then having a paraplegic and blind leader with a brain tumor and blood deficiency.

intentionally sandbagging your team seems like something that should be curbed or at least agreed upon by the people who have to deal with it. could care less for solo antags but not a fan of people taking purposefully debilitating traits on the team mode.

Rhials committed:
species-based outfit override registry

adds a list to the species datum, that works hand-in-hand with the new "equip species outfit"

Rhials commented:
I've pushed a solution that doesn't require any funky if checks on the outfit distribution side.

The current solution comes in two parts, the outfit_override_registry list and the equip_species_outfit proc.

The outfit_override_registry is a list, wherein the key is a default outfit path, and the value is the outfit you'd like to replace it with when equipping a species holder with the key outfit.

For instance, in the plasmaman registry the key/value pair is `/datum/outfit/syndicate/reinforcement = /datum/outfit/syndicate/reinforcement/plasmaman`, so when the reinforcement outfit is equipped via `equip_species_outfit()` proc, it will equip the `/datum/outfit/syndicate/reinforcement/plasmaman` outfit instead of the default.

This solution should be scalable to other species, and other use cases, without any issue.

Testing hasn't shown any differences in functionality -- this worked WAY better than I expected it to.

san7890 commented on code/modules/mob/living/carbon/human/inventory.dm:
```diff
@@ -385,6 +385,22 @@
+	else
+		outfit_to_equip = new outfit
+
+	if(!outfit_to_equip)
```
`isnull()` here

san7890 commented on code/modules/mob/living/carbon/human/species_types/plasmamen.dm:
```diff
@@ -51,6 +51,13 @@
+		/datum/outfit/syndicate = /datum/outfit/syndicate/plasmaman,
+		/datum/outfit/syndicate/leader = /datum/outfit/syndicate/leader/plasmaman,
+		/datum/outfit/syndicate/reinforcement = /datum/outfit/syndicate/reinforcement/plasmaman,
+		/datum/outfit/syndicate/full = /datum/outfit/syndicate/full/plasmaman,
```
alphabetize for better extensibility?

Rhials committed:
spare me the extensibilities

san7890 commented on code/modules/antagonists/nukeop/nukeop.dm:
```diff
@@ -35,9 +39,12 @@
 
-	operative.set_species(/datum/species/human) //Plasmamen burn up otherwise, and besides, all other species are vulnerable to asimov AIs. Let's standardize all operatives being human.
+	// If our nuke_ops_species pref is set to TRUE, (or we have no client) make us a human
+	if(!operative.client || operative.client.prefs.read_preference(/datum/preference/toggle/nuke_ops_species))
```
```suggestion
	if(isnull(operative.client) || operative.client.prefs.read_preference(/datum/preference/toggle/nuke_ops_species))
```

didn't see this my bad

san7890 committed:
Update code/modules/antagonists/nukeop/nukeop.dm

Livrah commented:
Can non human species get a macro bomb implant instead of regular one?
